app/models/order.rb

```diff
@@ -69,6 +69,17 @@ class Order < ActiveRecord::Base
 
   scope :goodcity_orders, -> { where(detail_type: 'GoodCity') }
 
+  def filter(opts = {})
+    query = FilterQueryBuilder.new
```

Comment:
@patrixr has moved the FilterQueryBuilder to another class file - can you rebase and update here?